docs/examples.md

```diff
@@ -0,0 +1,170 @@
+
+The package includes a class to convert WordPress block shortcodes to Wagtail blocks during the import, it's possible to use this class to transform content that is not a shortcode, but a specific HTML structure. This is done though the use of BeautifulSoup parsing the HTML rather than the provided regular expression that is part of a block shortcode handler.
+
+## How to implement a handler for a specific HTML structure
+
+The following handler will take some HTML input and convert all found specific HTML structures to a Wagtail block type.
+
+```python
+# anchor_image_handler.py
+
+
+from bs4 import BeautifulSoup
+
+from wagtail_wordpress_import.block_builder_defaults import get_or_save_image
+from wagtail_wordpress_import.prefilters.handle_shortcodes import (
+    BlockShortcodeHandler,
+    register,
+)
+
+
+@register()
+class AnchorImageHandler(BlockShortcodeHandler):
+    # shortcode name is required and is used to create a unique HTML tag.
+    shortcode_name = "anchor_image"
+    
+
+    def pre_filter(self, string):
+        # This method is required and will be called to transform a specific
+        # HTML structure found by BeautifulSoup into a custom HTML tag
+        # This example would produce an HTML tag of <wagtail_block_anchor_image>
+        soup = BeautifulSoup(string, "html.parser")
+        anchor_tags = soup.find_all("a")
+
+        anchors = []
+        for anchor in anchor_tags:
+            if anchor.find("img"):
+                anchors.append(anchor)
+
+        for a in anchors:
+            attrs = {}
+            attrs["data-src"] = a.find("img").attrs["src"]
+            attrs["data-caption"] = a.find("img").attrs["title"]
+            attrs["data-class"] = a.find("img").attrs["class"]
+            attrs["data-href"] = a.attrs["href"]
+            replacement_custom_tag = soup.new_tag(self.element_name)
+            replacement_custom_tag.attrs = attrs
+            a.replace_with(replacement_custom_tag)
+
+        return str(soup)
+
+    def construct_block(self, soup):
+        # get_or_save_image() is not included in the shortcode handler
+        # it is included in the package and can be used if required
+        image_file = get_or_save_image(soup.attrs["data-src"])
+
+        if image_file:
+            # The block type as a dict is returned to the BlockBuilder
+            return {
+                "type": "image",
+                "value": {
+                    "image": image_file.id,
+                    "caption": soup.attrs.get("data-caption"),
+                    "alignment": soup.attrs.get("data-class"),
+                    "link": soup.attrs.get("data-href"),
+                },
+            }
+        else:
+            # You should return at least one valid block here.
+            # This returns a raw_html block containing the custom HTML tag.
```

Comment:
**Suggestion**: Raw HTML blocks are not commonly used (at least by us). Provide a link to https://docs.wagtail.io/en/latest/reference/streamfield/blocks.html#wagtail.core.blocks.RawHTMLBlock to give context.